libraries/botbuilder-dialogs-adaptive/src/skillExtensions.ts

```diff
@@ -0,0 +1,39 @@
+/**
+ * @module botbuilder-dialogs-adaptive
+ */
+/**
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { DialogManager } from 'botbuilder-dialogs';
+import { BotFrameworkClient, SkillConversationIdFactoryBase } from 'botbuilder-core';
+
+declare module 'botbuilder-dialogs/lib/dialogManager' {
+    export interface DialogManager {
+        useSkillClient(skillClient: BotFrameworkClient): DialogManager;
+        useSkillConverationIdFactory(skillConverationIdFactory: SkillConversationIdFactoryBase): DialogManager;
+    }
+}
+
+export const skillClientKey = Symbol('SkillClient');
+
+/**
+ * Configures the skill client to use.
+ */
+DialogManager.prototype.useSkillClient = function(skillClient: BotFrameworkClient): DialogManager {
+    const _self = this as DialogManager;
+    _self.initialTurnState.set(skillClientKey, skillClient);
+    return _self;
+};
+
+export const skillConversationIdFactoryKey = Symbol('SkillConversationIdFactory');
+
+/**
+ * Configures the skill conversation id factory to use.
+ */
+DialogManager.prototype.useSkillConverationIdFactory = function(skillConversationIdFactory: SkillConversationIdFactoryBase): DialogManager {
```

Comment:
Per the sync-up on 7/2/20, this isn't the way to do extension methods in TS/JS as there _isn't_ a good way to port extension methods verbatim. (see microsoft/typescript#9, look for posts from **RyanCavanaugh**)

There are some non-trivial problems that we create by going down this path, problems that would be avoided through either inheritance (a new `"AdaptiveDialogManager"`) or standalone helper functions (example: [teamsActivityHelpers.ts](https://github.com/microsoft/botbuilder-js/blob/master/libraries/botbuilder/src/teamsActivityHelpers.ts)).

Additionally, our Skills and DialogManager concepts and implementation exist at a level outside of the Adaptive "namespace" and could be placed directly on the DialogManager. By placing these extension methods in an Adaptive package, we're likely going to confuse developers.

_(And we own the code base, so there will be additional confusion around our methods)_

### Workaround:

As a temporary workaround, @stevenic suggested keeping the keys and instead having developers set these initialTurnState values themselves.

```ts
import { skillConversationIdFactoryKey } from 'botbuilder-dialogs-adaptive'; 
import { DialogManager } from 'botbuilder-dialogs';

const dm = new DialogManager(/* some dialog */);
dm.initialTurnState.set(skillConversationIdFactoryKey, /* some idFactory */);
```

For programmatically registering values onto the DialogManager.initialTurnState, the code here is lightweight enough to implement via the application that uses Adaptive libraries as its runtime.